crates/executor/src/executor.rs

```diff
@@ -319,23 +320,22 @@ pub fn execute<DB: StateProvider>(
         revm_wrap::fill_tx_env(&mut evm.env.tx, transaction);
 
         // Execute transaction.
-        let (revm::ExecutionResult { exit_reason, gas_used, logs, .. }, state) = evm.transact();
+        let (revm::ExecutionResult { exit_reason, gas_used, logs, gas_refunded, .. }, state) =
+            evm.transact();
+
+        tracing::trace!(target:"evm","Executing transaction {:?}, gas:{gas_used} refund:{gas_refunded}",transaction.hash());
 
         // Fatal internal error.
         if exit_reason == revm::Return::FatalExternalError {
             return Err(Error::ExecutionFatalError)
         }
 
         // Success flag was added in `EIP-658: Embedding transaction status code in receipts`.
-        let is_success = matches!(
-            exit_reason,
-            revm::Return::Continue |
-                revm::Return::Stop |
-                revm::Return::Return |
-                revm::Return::SelfDestruct
-        );
-
-        // TODO add handling of other errors
+        let is_success = match exit_reason {
+            revm::return_ok!() => true,
+            revm::return_revert!() => false,
+            e => return Err(Error::EVMError { error_code: e as u32 }),
```

Comment:
TODO: This error code should be typed

Reply:
https://github.com/paradigmxyz/reth/issues/463